docs/components/modeler/bpmn/user-tasks/user-tasks.md

```diff
@@ -100,18 +84,82 @@ import ISO8601DateTime from '../assets/react-components/iso-8601-date-time.md'
+   of the task's `zeebe:formDefinition` extension element (see the [XML representation](#camunda-form-linked)).
+   Form ID references always refer to the latest deployed version of the Camunda Form.
+
+   You can read more about Camunda Forms in the [Camunda Forms guide](/guides/utilizing-forms.md) or the [Camunda Forms reference](/docs/components/modeler/forms/camunda-forms-reference.md)
+   to explore all configuration options for form elements.
+
+2. A **custom form reference** can specify any custom identifier in the user task using the `externalReference`
+   attribute of the task's `zeebe:formDefinition` extension element (see the [XML representation](#camunda-form-linked)).
+   How the identifier is interpreted depends on your implementation.
+   You can use it to associate a custom form, route to a custom application, or a URL to a web page, for example.
+   A custom form reference will not be shown in Tasklist.
+
+:::info
+For user tasks with a [job worker implementation](#job-worker-implementation), the custom form references are defined on the `formKey` attribute
+of the `zeebe:formDefinition` extension element instead of the `externalReference` attribute.
+
+Furthermore, there is third form option for job worker-based user tasks: embedded Camunda forms. You can use them to
+embed a form's JSON configuration directly into the BPMN process XML as a `zeebe:UserTaskForm` extension element of the
+process element.The embedded form can then be referenced via the `formKey` attribute (see [XML representation](#camunda-form-embedded)).
+:::
+
+## Job worker implementation
+
+A user task does not have to be managed as a Zeebe user task. Instead, you can also use
+job workers to implement your user task logic.
+
+You can define a job worker implementation for a user task by removing its `zeebe:userTask` extension element.
+
+User tasks with a job worker implementation behave exactly like [service tasks](/components/modeler/bpmn/service-tasks/service-tasks.md). The differences between these task
+types are the visual representation (i.e. the task marker) and the semantics for the model.
+
+When a process instance enters a user task with job worker implementation, it creates
+a corresponding job and waits for its completion. A job worker should request jobs of the job type `io.camunda.zeebe:userTask`
+and process them. When the job is completed, the process instance continues.
+
+Use [task headers](/components/modeler/bpmn/service-tasks/service-tasks.md#task-headers) to pass static parameters to the job
+worker.
 
-## Task headers
+Define [variable mappings](/components/concepts/variables.md#inputoutput-variable-mappings)
+the [same way as a service task does](/components/modeler/bpmn/service-tasks/service-tasks.md#variable-mappings)
+to transform the variables passed to the job worker, or to customize how the variables of the job merge.
 
-A user task can define an arbitrary number of `taskHeaders`; they are static
```

Comment:
> A user task can define an arbitrary number of `taskHeaders`. Task headers are static metadata for your custom implementation. The headers will be ignored by Zeebe. 

Shouldn't we copy that to the updated docs as well, since you also implemented it?

Reply:
Yeah true, good catch 👍 🙂